pytorch_pfn_extras/onnx/_grad.py

```diff
@@ -0,0 +1,98 @@
+from typing import Generator, List, Optional, Tuple
+from contextlib import contextmanager
+
+import torch
+import torch.onnx
+import threading
+from pytorch_pfn_extras.onnx._as_output import as_output
+
+
+_grad_state = threading.local()
+
+
+@contextmanager
+def init_grad_state() -> Generator[None, None, None]:
+    _grad_state.n_grad_call = 0
+    try:
+        yield
+    finally:
+        _grad_state.n_grad_call = None
+
+
+def grad(
+    output: torch.Tensor,
+    inputs: Tuple[torch.Tensor, ...],
+    retain_graph: Optional[bool] = None,
+    create_graph: bool = False,
+    allow_unused: bool = False,
+    only_inputs: bool = True,
+) -> Tuple[Optional[torch.Tensor], ...]:
+    grad_output = torch.ones_like(output)
+
+    if torch.jit.is_tracing():
+        assert hasattr(_grad_state, "n_grad_call")
```

Comment:
Maybe better to raise an error with friendly message (e.g., `ppe.onnx.grad() can only be used in conjunction with export functions under `ppe.onnx`) instead of just an assertion?